examples/calculator/main.cpp

```diff
@@ -15,27 +15,26 @@
 #include "MultiplicationModel.hpp"
 #include "DivisionModel.hpp"
 
-static bool
+static std::unique_ptr<DataModelRegistry>
```

Comment:
I made some comments from my phone but can't see them now.

What is the benefit of using `unique_ptr` ? 
I would return just `DataModelRegistry`